types/session_price.yaml

```diff
@@ -0,0 +1,73 @@
+description: Types to provide the session price
+types:
+  SessionCostChunk:
+    description: >-
+      A chunk of the session cost. The total cost of the session is the sum of
+      all chunks.
+    type: object
+    additionalProperties: false
+    properties:
+      timestamp_from:
+        description: >-
+          Absolute timestamp for the start of this chunk in RFC3339 UTC format
+        type: string
+        format: date-time
+      timestamp_to:
+        description: >-
+          Absolute timestamp for the end of this chunk in RFC3339 UTC format
+        type: string
+        format: date-time
+      cost_energy:
+        description: >-
+          Cost of the energy consumed during this chunk in the currency
+          specified in the session price
+        type: number
+      cost_time:
```

Comment:
Can we make it more generic? How about we  call it additional_cost and users may include here what ever they want?

Reply:
or even more generic - something like
```cpp
struct Chunk{
   std::string timestamp_from;
   std::string timestamp_to;
  float cost;
  std::string purpose;
};
```

Reply:
never use floating point for money ever

https://www.linkedin.com/pulse/never-use-floating-point-double-data-types-monetary-elkhawaldeh/

Reply:
I've refactored it a bit. Can you take a look?